R/threshold_perf.R

```diff
@@ -166,30 +184,31 @@ threshold_perf.data.frame <- function(.data,
     .data <- .data %>% dplyr::group_by(.threshold)
   }
 
-  two_class <- yardstick::metric_set(sens, spec, j_index)
 
-  .data_metrics <- two_class(
+  .data_metrics <- metrics(
     .data,
     truth = truth,
     estimate = alt_pred,
     event_level = event_level
   )
 
-  # Create the `distance` metric data frame
-  # and add it on
-  sens_vec <- .data_metrics %>%
-    dplyr::filter(.metric == "sens") %>%
-    dplyr::pull(.estimate)
-
-  dist <- .data_metrics %>%
-    dplyr::filter(.metric == "spec") %>%
-    dplyr::mutate(
-      .metric = "distance",
-      # .estimate is spec currently. this recodes as distance
-      .estimate = (1 - sens_vec) ^ 2 + (1 - .estimate) ^ 2
-    )
+  if (measure_sens_spec) {
+    # Create the `distance` metric data frame
+    # and add it on
+    sens_vec <- .data_metrics %>%
+      dplyr::filter(.metric == "sens") %>%
+      dplyr::pull(.estimate)
+
+    dist <- .data_metrics %>%
+      dplyr::filter(.metric == "spec") %>%
+      dplyr::mutate(
+        .metric = "distance",
+        # .estimate is spec currently. this recodes as distance
```

Comment:
Sorry, I don't understand this comment

Reply:
We duplicated the specificity results (the `"spec"`) in the comment.